library/spdm_requester_lib/libspdm_req_encap_request.c

```diff
@@ -235,13 +235,39 @@ libspdm_return_t libspdm_encapsulated_request(libspdm_context_t *spdm_context,
             libspdm_release_receiver_buffer (spdm_context);
             return status;
         }
-        if (libspdm_encapsulated_request_response->header.request_response_code !=
-            SPDM_ENCAPSULATED_REQUEST) {
+        if (libspdm_encapsulated_request_response->header.spdm_version !=
+            spdm_get_encapsulated_request_request->header.spdm_version) {
             libspdm_release_receiver_buffer (spdm_context);
             return LIBSPDM_STATUS_INVALID_MSG_FIELD;
         }
-        if (libspdm_encapsulated_request_response->header.spdm_version !=
-            spdm_get_encapsulated_request_request->header.spdm_version) {
+        if (libspdm_encapsulated_request_response->header.request_response_code == SPDM_ERROR) {
+            const uint8_t error_code = libspdm_encapsulated_request_response->header.param1;
+
+            libspdm_release_receiver_buffer(spdm_context);
+
+            if (spdm_response_size < sizeof(spdm_error_response_t)) {
+                libspdm_release_receiver_buffer (spdm_context);
+                return LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            }
+
+            if (mut_auth_requested == 0) {
```

Comment:
Is that enough?

Should we also consider basic_mut_auth here?

Reply:
`mut_auth_requested == 0` is also used for the basic mutual authentication flow. In particular only digests, certificates, and challenge should be allowed in the basic mutual authentication flow, but it does not look like libspdm explicitly enforces that. I will file a separate issue for that.